include/nbl/core/containers/DoublyLinkedList.h

```diff
@@ -0,0 +1,330 @@
+		// After address allocator creation we can free the old buffer
+		_NBL_ALIGNED_FREE(m_reservedSpace);
+		m_cap = newCapacity;
+		m_array = newArray;
+		m_reservedSpace = newReservedSpace;
+		
+
+		return true;
+	}
+
+	//Constructor, capacity determines the amount of allocated space
+	DoublyLinkedList(const uint32_t capacity, disposal_func_t&& dispose_f = disposal_func_t()) : m_dispose_f(std::move(dispose_f))
+	{
+		const auto firstPart = core::alignUp(address_allocator_t::reserved_size(1u, capacity, 1u), alignof(node_t));
+		m_reservedSpace = _NBL_ALIGNED_MALLOC(firstPart + capacity * sizeof(node_t), alignof(node_t));
+		m_array = reinterpret_cast<node_t*>(reinterpret_cast<uint8_t*>(m_reservedSpace) + firstPart);
+
+		m_addressAllocator = std::make_unique<address_allocator_t>(m_reservedSpace, 0u, 0u, 1u, capacity, 1u);
+		m_cap = capacity;
+		m_back = invalid_iterator;
+		m_begin = invalid_iterator;
+	}
+
+	DoublyLinkedList() = default;
+
+	DoublyLinkedList(const DoublyLinkedList& other) = delete;
+
+	DoublyLinkedList& operator=(const DoublyLinkedList& other) = delete;
+
+	DoublyLinkedList& operator=(DoublyLinkedList&& other)
+	{
+		m_addressAllocator = std::move(other.m_addressAllocator);
+		m_reservedSpace = other.m_reservedSpace;
+		m_array = other.m_array;
+		m_dispose_f = std::move(other.m_dispose_f);
+		m_cap = other.m_cap;
+		m_back = other.m_back;
+		m_begin = other.m_begin;
+
+		// Nullify other
+		other.m_addressAllocator = nullptr;
+		other.m_reservedSpace = nullptr;
+		other.m_array = nullptr;
+		other.m_cap = 0u;
+		other.m_back = 0u;
+		other.m_begin = 0u;
+		return *this;
+	}
+
+	~DoublyLinkedList()
+	{
+		disposeAll();
+		_NBL_ALIGNED_FREE(m_reservedSpace);
+	}
+
+private:
+	//allocate and get the address of the next free node
+	inline uint32_t reserveAddress()
+	{
+		uint32_t addr = m_addressAllocator->alloc_addr(1u, 1u);
+		return addr;
+	}
+
+	//create a new node which stores data at already allocated address, 
+	inline void insertAt(uint32_t addr, value_t&& val)
+	{
+		assert(addr < m_cap);
+		assert(addr != invalid_iterator);
+		SDoublyLinkedNode<Value>* n = new(m_array + addr) SDoublyLinkedNode<Value>(std::move(val));
+		n->prev = invalid_iterator;
+		n->next = m_begin;
+
+		if (m_begin != invalid_iterator)
+			getBegin()->prev = addr;
+		if (m_back == invalid_iterator)
+			m_back = addr;
+		m_begin = addr;
+	}
+
+	/**
+	* @brief Calls disposal function on all elements of the list.
+	*/
+	inline void disposeAll()
+	{
+		if (m_dispose_f && m_begin != invalid_iterator)
+		{
+			auto* begin = getBegin();
+			auto* back = getBack();
+			while (begin != back)
+			{
+				m_dispose_f(begin->data);
+				begin = get(begin->next);
+			}
+			m_dispose_f(back->data);
+		}
+	}
+
+	inline void common_delete(uint32_t address)
+	{
+		if (m_dispose_f)
+			m_dispose_f(get(address)->data);
+		get(address)->~node_t();
```

Comment:
style hint, use `std::uninitialized_construct` and destruct instead of placement new and explicit destructor invoke, see the commandbuffer/pool stuff

Reply:
or use a true C++11 style `allocator_traits` and its construct/destruct methods https://github.com/Devsh-Graphics-Programming/Nabla/pull/823/files#r1979485663